src/protocols/ProtocolEthernet.cpp

```diff
@@ -0,0 +1,60 @@
+#include "ProtocolEthernet.h"
+#include <unistd.h>
+#include <sstream>
+
+namespace RACprotocol
+{
+    ProtocolEthernet::ProtocolEthernet() : pProtoStruct(nullptr)
+    {
+	pProtoStruct = new protocol;
+	bzero(pProtoStruct, sizeof(*pProtoStruct));
+    }
+
+    ProtocolEthernet::~ProtocolEthernet()
+    {
+	if (pProtoStruct != nullptr)
+	{
+	    delete pProtoStruct;
+	}
+    }
+
+    std::string		ProtocolEthernet::getMACFormated(const uint8_t *uMAC) const
+    {
+	char	cMAC[64];
```

Comment:
You could use `std::stringstream` in combination with `set_precision` if I remember correctly

Reply:
maybe its work with std::stringstream but its work well with this method so i will keep it